t/plugin/limit-count-redis3.t

```diff
@@ -186,3 +186,63 @@ passed
     }
 --- response_body
 ["1","0","0"]
+
+
+
+=== TEST 5: set route, with redis host, port and right username and password
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "uri": "/hello",
+                    "plugins": {
+                        "limit-count": {
+                            "count": 2,
+                            "time_window": 60,
+                            "rejected_code": 503,
+                            "key": "remote_addr",
+                            "policy": "redis",
+                            "redis_host": "127.0.0.1",
+                            "redis_port": 6379,
+                            "redis_timeout": 1001,
+                            "redis_username": "alice",
```

Comment:
I'm very confused about here, does redis accept the following 2 forms at the same time? No configuration required?
1. redis_username and redis_password
2. only redis_password

Reply:
Yes, the two forms are accepted. It uses the module openresty/lua-resty-redis based on the Redis Auth command : https://github.com/openresty/lua-resty-redis#redis-authentication. 
The Redis Auth command accepted only the password or the username and password : https://redis.io/commands/auth/

Reply:
It looks like that you don't config the redis:
https://github.com/apache/apisix/blob/master/t/plugin/limit-count-redis.t#L222
Could you add the following test cases?
1. Using the wrong username and the right password
2. Using the right username and the wrong password

Reply:
i add it into limit-count-redis4.t

Reply:
Could the three test cases be put together? THX

Reply:
Why are the test cases from `TEST 5` to `TEST 7` in `limit-count-redis3.t` not moved to `limit-count-redis.t`? They tested the same features as the ones you moved

Reply:
Because i don't write it, they were already there. I change only the code i write for the least impact.

Reply:
I hope it is good now